Bugsnag/KSCrash/Source/KSCrash/Recording/BSG_KSCrashReport.c

```diff
@@ -926,7 +928,19 @@ void bsg_kscrw_i_writeThread(const BSG_KSCrashReportWriter *const writer,
         writer->addBooleanElement(writer, BSG_KSCrashField_Crashed,
                                   isCrashedThread);
         writer->addBooleanElement(writer, BSG_KSCrashField_CurrentThread,
-                                  thread == bsg_ksmachthread_self());
+                                  isSelfThread);
+
+        // Fetching the current thread name is only safe as of libpthread-330.201.1
+        // which was used in ios+tvos 12 and macos 10.14
+        if (bsgos_available(BSGOS_IOS, 12, 0) ||
+            bsgos_available(BSGOS_TVOS, 12, 0) ||
+            bsgos_available(BSGOS_MACOS, 10, 14)) {
```

Comment:
The complexity and binary size impact of the `bsgos_available` function(s) (760 bytes) feels a bit unjustified for this use.

How about a much simpler approach - checking the version of CoreFoundation to deduce the OS version? It was a commonly used approach before introduction of the `@available` intrinsic.

```suggestion
        if (kCFCoreFoundationVersionNumber >= kCFCoreFoundationVersionNumber_iOS_12_0) {
```

Annoyingly, Apple stopped adding defines for the version after iOS 9, so we'll need to declare this ourselves:

```
// Reference: http://iphonedevwiki.net/index.php/CoreFoundation.framework
#define kCFCoreFoundationVersionNumber_iOS_12_0 793.00
```